lib/log.c

```diff
@@ -212,6 +217,7 @@ void vzlog(int priority, const char *format, va_list args)
 
 		/* In this case we return at here. */
 		errno = original_errno;
+		pthread_mutex_unlock(&loglock);
```

Comment:
This will ensure that fprintfs happen sequentially across all threads, while this will yield nicer log files, it does yield longer than needed locks.  The alternative is to move the unlock to line 211.  What do you think?

Reply:
I think not mangling logs is more important than the instructions saved. After all, `loglock` contention should be relatively rare.

Reply:
the issue is timing/delay in execution (due to lock while printing) vs pretty log - not instruction count which has no change.

Reply:
I misspoke and edited my comment about 30 seconds after posting it. Like I said I honestly don't think this is worth the micro optimization -- I expect the time that two threads are trying to log at the same time to be pretty small -- I would rather have readable logs